sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/AbstractRule.java

```diff
@@ -24,10 +24,16 @@
 public abstract class AbstractRule implements Rule {
 
     /**
-     * rule id.
+     * Rule id
      */
     private Long id;
 
+    private String app;
+
+    private String ip;
```

Comment:
It seems perplexing to associate a rule entity with IP/port... If it's needed in Sentinel dashboard, maybe we could have a copy of it in the dashboard module?

Reply:
Thanks! @sczyh30 
These fields(app,ip,port) have been removed and I found a better way for `sentinel-dashboard-repository-memory`,by improving SystemRuleManager, store and manage rule id when modifying and getting rules.